### ISSUE TITLE
Recursive Extraction Checkbox

### ISSUE THREAD

**Issue by speck-drum** [aonez/Keka#337]
If checked, after processing an archive Keka would scan the extracted files for archives and replace every found archive by its extracted contents which is scanned for archives, too … and so on.

This solves the issue of .tar.gz, .tgz, tar.bzip, … archives in a more generalized way.

**Comment by aonez** [comment]
Current tarball packages are extracted automatically, but what you're here suggesting goes beyond that. So for example for an extracted ZIP file that contains a plain txt and a folder containing two TGZ tar balls, you say Keka should extract those too?

**Comment by speck-drum** [comment]
Yes, this is what I suggested (it was a feature of Stuffit Expander). But at the moment I am not sure whether there exists a use case for this feature. Probably a checkbox “Repeat expanding as long as the result is a single file” would do a better job.

**Comment by ravenium** [comment]
Came here to search, noticed this ticket, nodded :)

My use case is somewhat related.  I work with a lot of .tar.bz2's and .tar.gz's, so not having to decompress twice is key. What's weird is that Keka can only "double extract" when the filename is explicitly .tar.bz2 or .tar.gz, not just .bz2 or .gz.

Normally I wouldn't complain as analyzing files is probably resource expensive and not always desired, but OSX's native Archive utility can do this regardless of extension.  I'm wondering if this is an explicit exception case needed since it's pretty common.

**Comment by aonez** [comment]
Just yesterday downloaded a `file.tar.gz` twice and Safari renamed the second one to `file.tar 2.gz`... So the extraction resulted in `file.tar 2`.

So this will be implemented:

- If extraction results in a single file
  - Check if that file can be extracted
    - Always extract if it's a TAR
    - For other compatible formats, if "Recursively extract..." is enabled extract it
